src/logic/hooks/useEstimateTransactionGas.tsx

```diff
@@ -0,0 +1,114 @@
+  txConfirmations === threshold || !!preApprovingOwner || threshold === 1
+
+type UseEstimateTransactionGasProps = {
+  txData: string
+  txRecipient: string
+  txConfirmations?: number
+  txAmount?: string
+  preApprovingOwner?: string
+  operation?: number
+}
+
+type TransactionGasEstimationResult = {
+  txEstimationExecutionStatus: EstimationStatus
+  gasEstimation: number // Amount of gas needed for execute or approve the transaction
+  gasCost: string // Cost of gas in raw format (estimatedGas * gasPrice)
+  gasCostFormatted: string // Cost of gas in format '< | > 100'
+  gasPrice: string // Current price of gas unit
+  isExecution: boolean // Returns true if the user will execute the tx or false if it just signs it
+}
+
+export const useEstimateTransactionGas = ({
+  txRecipient,
+  txData,
+  txConfirmations,
+  txAmount,
+  preApprovingOwner,
+  operation,
+}: UseEstimateTransactionGasProps): TransactionGasEstimationResult => {
+  const [gasEstimation, setGasEstimation] = useState<TransactionGasEstimationResult>({
+    txEstimationExecutionStatus: EstimationStatus.LOADING,
+    gasEstimation: 0,
+    gasCost: '0',
+    gasCostFormatted: '< 0.001',
+    gasPrice: '0',
+    isExecution: false,
+  })
+  const { nativeCoin } = getNetworkInfo()
+  const safeAddress = useSelector(safeParamAddressFromStateSelector)
+  const threshold = useSelector(safeThresholdSelector)
+
+  useEffect(() => {
+    let isCurrent = true
+
+    const estimateGas = async () => {
+      if (!txData.length) {
+        return
+      }
+
+      try {
+        const isExecution = checkIfTxIsExecution(Number(threshold), preApprovingOwner, txConfirmations)
+
+        const gasEstimation = await estimateTransactionGas({
+          safeAddress,
+          txRecipient,
+          txData,
+          txAmount,
+          isExecution,
+          operation,
+        })
+        const gasPrice = await calculateGasPrice()
+        const estimatedGasCosts = gasEstimation * parseInt(gasPrice, 10)
+        const gasCost = fromTokenUnit(estimatedGasCosts, nativeCoin.decimals)
+        const gasCostFormatted = formatAmount(gasCost)
+        if (isCurrent) {
+          setGasEstimation({
+            txEstimationExecutionStatus: gasEstimation <= 0 ? EstimationStatus.FAILURE : EstimationStatus.SUCCESS,
+            gasEstimation,
+            gasCost,
+            gasCostFormatted,
+            gasPrice,
+            isExecution,
+          })
+        }
+      } catch (error) {
+        // We put a fixed the amount of gas to let the user try to execute the tx, but it's not accurate so it will probably fail
+        const gasEstimation = 10000
+        const gasCost = fromTokenUnit(gasEstimation, nativeCoin.decimals)
+        const gasCostFormatted = formatAmount(gasCost)
+        setGasEstimation({
+          txEstimationExecutionStatus: EstimationStatus.FAILURE,
+          gasEstimation,
+          gasCost,
+          gasCostFormatted,
+          gasPrice: '1',
```

Comment:
just curious, why not -1?  I think it's more standard

Reply:
To avoid a negative cost that could confuse the user, maybe 0 it's better